core/drivers/stm32_gpio.c

```diff
@@ -1198,8 +1290,30 @@ static TEE_Result stm32_pinctrl_conf_apply(struct pinconf *conf)
 {
 	struct stm32_pinctrl_array *ref = conf->priv;
 	struct stm32_pinctrl *p = ref->pinctrl;
+	struct stm32_gpio_bank *bank = NULL;
+	TEE_Result res = TEE_ERROR_GENERIC;
 	size_t pin_count = ref->count;
 	size_t n = 0;
+	bool error = false;
+
+	for (n = 0; n < pin_count; n++) {
+		bank = stm32_gpio_get_bank(p[n].bank);
+		res = acquire_rif_semaphore_if_needed(bank, p[n].pin);
+		if (res) {
+			EMSG("Failed to acquire GPIO %c%u semaphore",
+			     bank->bank_id + 'A', p[n].pin);
+			error = true;
+		}
+	}
+
+	if (error) {
```

Comment:
If an error occured in acquire_rif_semaphore_if_needed(), then the semaphore was not taken by the OP-TEE. I think this can be removed

Reply:
We still have to release the other acquired semaphores.

Reply:
ok